src/primval.rs

```diff
@@ -14,6 +14,8 @@ pub enum PrimVal {
 
     Ptr(Pointer),
     FnPtr(Pointer),
+    VtablePtr(Pointer, Pointer),
+    SlicePtr(Pointer, u64),
```

Comment:
@eddyb Seeing as we already argued about this in IRC, what do you think about this commit? I'm still in the middle of reading it over and figuring out if there are any issues with this compared to `Value::ByValPair`. Remember that `Pointer` supports integer pointers now.

For one thing, this won't automatically be able to work with other primval pairs like the result of checked binary operations.